### DISCUSSION TITLE
why the 2 in flate2?

### DISCUSSION THREAD

**Discussion by oriongonza** [rust-lang/flate2-rs#377] [Q&A] [answered]
This has bothered me from the start, what's the story behind that name?

**Comment by oyvindln** [comment, marked as answer]
There [used](https://github.com/rust-lang/rust/tree/1.19.0/src/libflate) to be a simple library that was a simple wrapper around the miniz C library in the main rust source code called "flate" which meant that was also a reserved library name. So, presumably @alexcrichton (or whoever else started it) decided to simply attach a 2 at the end for this to not conflict with it when flate2 was originally started. I believe the rust compiler switched to using flate2 from version 1.20.

**Reply by alexcrichton** [reply]
Yep that's exactly why :)
